File: test-suite-kotlin/src/test/kotlin/io/micronaut/views/docs/soy/SoyRenamingTest.kt
package io.micronaut.views.docs.soy

import io.micronaut.context.annotation.Property
import io.micronaut.core.util.StringUtils
import io.micronaut.http.client.HttpClient
import io.micronaut.http.client.annotation.Client
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

@Property(name = "micronaut.security.enabled", value = StringUtils.FALSE)
@Property(name = "spec.name", value = "soy")
@Property(name = "micronaut.views.soy.renaming-enabled", value = StringUtils.TRUE)
@Property(name = "micronaut.views.velocity.enabled", value = StringUtils.FALSE)
@MicronautTest
internal class SoyRenamingTest {
    @Inject
    @field:Client("/")
    lateinit var httpClient: HttpClient

    @Test
    fun cssClassesAreRenamedWithTheRenamingMap() {
        val html = httpClient.toBlocking().retrieve("/soy/renaming")
        assertTrue(html.contains("<div class=\"a-b-c a-b-c-d\">Renamed</div>"), html)
    }
}

File: test-suite-python/src/test/python/micronaut/views/docs/soy/SoyRenamingTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@Property(name="micronaut.security.enabled", value="false")
@Property(name="spec.name", value="soy")
@Property(name="micronaut.views.soy.renaming-enabled", value="true")
@Property(name="micronaut.views.velocity.enabled", value="false")
@MicronautTest
class SoyRenamingTest:
    httpClient: Annotated[HttpClient, Inject, Client("/")]

    @Test
    def test_css_classes_are_renamed_with_the_renaming_map(self):
        html = self.httpClient.toBlocking().retrieve("/soy/renaming")
        assert '<div class="a-b-c a-b-c-d">Renamed</div>' in html, html
